### PR TITLE
Restructure maps to live in heierarchy under docs.code.org instead of singletons under a given course

### PR DESCRIPTION
**Pull request by mrjoshida**
Reopening using django page tree logic to organize concept maps

### CODE REVIEW

**Comment by mrjoshida** [comment]
@Hamms I took another whack at moving the maps to documentation with a simpler url schema. This assumes a top level map with the slug "concepts" with other maps hung underneath. Eg:

image

Design could still use some love, but I want to get this out so we can fix some maps that have typos using the new url structure

image

File: curriculumBuilder/settings.py [hunk 1]
```diff
@@ -50,11 +50,16 @@
...
+    # Used to generate the main curriculum.code.org course list
     (1, _("Main Curriculum Listing"), "pages/menus/dropdown.html"),
-    (2, _("Unused"), "pages/menus/tree.html"),
-    (3, _("Unused"), "pages/menus/footer.html"),
+    # Used to generate the docs.code.org/concepts/ menu
+    (2, _("Concept Maps"), "pages/menus/tree.html"),
+    #(3, _("Unused"), "pages/menus/footer.html"),
 )
 
+# Don't add new pages to any menuds by default
```
Comment (Hamms) [inline curriculumBuilder/settings.py]: nit: s/menuds/menus

File: documentation/models.py [hunk 2]
```diff
@@ -327,12 +325,14 @@ def publish(self, children=False):
...
                 logger.exception('Failed to publish %s' % self)
 
+    '''
     def save(self, *args, **kwargs):
 
         if not self.slug:
             self.slug = slugify(self.title)[:255]
 
         super(Map, self).save(*args, **kwargs)
+    '''
```
Comment (Hamms) [inline documentation/models.py]: should this just be removed?

File: documentation/templates/documentation/map.html [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+{% load staticfiles %}
+{% load mezzanine_tags %}
+{% load keyword_tags %}
+
+{% block meta_keywords %}{% metablock %}
+    {% keywords_for map as keywords %}
+    {% for keyword in keywords %}
+        {% if not forloop.first %}, {% endif %}
+        {{ keyword }}
+    {% endfor %}
```
Comment (Hamms) [inline documentation/templates/documentation/map.html]: is there a reason this isn't just a `join`?

File: documentation/urls.py [hunk 4]
```diff
@@ -6,10 +6,11 @@
 urlpatterns = [
     multiurl(
         # url(r'^$', views.index, name='home'),
+        url(r'^(?P<slug>.*)/$', views.map_view, name="map_view"),
+        # url(r'^concepts/$', views.maps_view, name='maps_view'),
+        # url(r'^concepts(?P<parents>.*)/(?P<slug>[-\w.]+)/$', views.page_view, name='page_view'),
```
Comment (Hamms) [inline documentation/urls.py]: should these be removed?

File: documentation/templates/documentation/map.html (hunk 3 above)
Comment (mrjoshida) [reply documentation/templates/documentation/map.html]: That's the pattern mezzanine uses everywhere else, the join template tag definitely would be cleaner

File: documentation/urls.py (hunk 4 above)
Comment (mrjoshida) [reply documentation/urls.py]: Yup

File: documentation/models.py (hunk 2 above)
Comment (mrjoshida) [reply documentation/models.py]: Yup

**Pull request merged**